### PR TITLE
Refactor memberships and pages

### PR DESCRIPTION
**Pull request by eikhr**
# Description

This PR does a few different changes that are highly related, so difficult to split up into separate PRs.
1. Refactors `memberships` redux code to use `createSlice`, `legoAdapter` and simpler selectors.
2. Fully types the "pages"-components, including the different page-renderers for groups or normal flatpages that were previously untyped/wrongly typed.
3. Some minor changes to the interest group detail page.

# Result

Type-safety, removed 17 type errors and a lot of `any`:)

Very minor functional changes. 
- On the interest group detail page, I removed the weird [fallback logo](https://imgur.com/Is9VKjb) and added a loading indicator.

# Testing

- [x] I have thoroughly tested my changes.

I have tested:
- All types of pages. Flatpages, group-pages, 404-page and "info-om-abakus".
- Adding, removing and changing roles in groups. 
- Joining and leaving interest groups

---

ABA-688

### CODE REVIEW

File: app/routes/pages/components/PageEditor.tsx [hunk 1]
```diff
@@ -53,11 +55,11 @@ type FormValues = Omit<ApiRequestBody, 'category'> & {
...
 const PageEditor = () => {
-  const { pageSlug } = useParams<{ pageSlug: string }>();
+  const { pageSlug } = useParams<PageDetailParams>() as PageDetailParams;
   const page = useAppSelector((state) =>
-    selectPageBySlug(state, {
-      pageSlug,
-    }),
+    pageSlug
+      ? (selectPageBySlug(state, pageSlug) as AuthDetailedPage)
+      : undefined,
```
Comment (ivarnakken) [inline app/routes/pages/components/PageEditor.tsx:62]: Is this not handled by `selectPageBySlug`?

File: app/routes/admin/groups/components/GroupPage.tsx [hunk 2]
```diff
@@ -58,9 +60,14 @@ const GroupPageNavigation = ({
...
 
+export type GroupPageParams = {
+  groupId: string;
+};
 const GroupPage = () => {
-  const { groupId } = useParams<{ groupId?: string }>();
-  const group = useAppSelector((state) => selectGroupById(state, groupId!));
+  const { groupId } = useParams<Optional<GroupPageParams>>(); // optional because of the /admin/groups route with no groupId
+  const group = useAppSelector((state) =>
+    groupId ? (selectGroupById(state, groupId) as DetailedGroup) : undefined,
```
Comment (ivarnakken) [inline app/routes/admin/groups/components/GroupPage.tsx:69]: Handled by `selectGroupById`?

File: app/routes/admin/groups/components/GroupPermissions.tsx [hunk 3]
```diff
@@ -122,8 +123,10 @@ const PermissionList = ({ group }: PermissionListProps) => {
 };
 
 const GroupPermissions = () => {
-  const { groupId } = useParams<{ groupId: string }>();
-  const group = useAppSelector((state) => selectGroupById(state, groupId!));
+  const { groupId } = useParams<GroupPageParams>() as GroupPageParams;
+  const group = useAppSelector(
+    (state) => selectGroupById(state, groupId) as DetailedGroup | undefined,
+  );
```
Comment (ivarnakken) [inline app/routes/admin/groups/components/GroupPermissions.tsx:129]:
```suggestion
  const group = useAppSelector(
    (state) => selectGroupById<DetailedGroup>(state, groupId)
  );
```

File: app/routes/errors/HTTPError.tsx [hunk 4]
```diff
@@ -14,8 +14,8 @@ const HTTPMapping = {
 };
 const fallbackStatus = 404;
 
-const getHTTPError = (statusCode) =>
-  HTTPMapping[statusCode] || HTTPMapping[fallbackStatus];
+const getHTTPError = (statusCode?: number) =>
+  HTTPMapping[statusCode || fallbackStatus] || HTTPMapping[fallbackStatus];
```
Comment (ivarnakken) [inline app/routes/errors/HTTPError.tsx:18]:
```suggestion
const getHTTPError = (statusCode?: number) =>
  HTTPMapping[statusCode || fallbackStatus];
```
Is this not the same?

File: app/routes/interestgroups/components/InterestGroupDetail.tsx [hunk 5]
```diff
@@ -112,19 +110,21 @@ const Contact = ({ group }: { group: Group }) => {
...
+};
 const InterestGroupDetail = () => {
-  const { groupId } = useParams();
-  const selectedGroup = useAppSelector((state) =>
-    selectGroupById(state, groupId!),
+  const { groupId } =
+    useParams<InterestGroupDetailParams>() as InterestGroupDetailParams;
+  const group = useAppSelector(
+    (state) =>
+      selectGroupById(state, groupId) as PublicDetailedGroup | undefined,
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroupDetail.tsx:121]:
```suggestion
  const group = useAppSelector(
    (state) =>
      selectGroupById<PublicDetailedGroup>(state, groupId)
```

**Review by ivarnakken — APPROVED** [review]

File: app/routes/pages/components/PageEditor.tsx (hunk 1 above)
Comment (eikhr) [reply app/routes/pages/components/PageEditor.tsx]: It is after rebasing😁

File: app/routes/admin/groups/components/GroupPage.tsx (hunk 2 above)
Comment (eikhr) [reply app/routes/admin/groups/components/GroupPage.tsx]: Yup

File: app/routes/errors/HTTPError.tsx (hunk 4 above)
Comment (eikhr) [reply app/routes/errors/HTTPError.tsx:18]: Oh, hahah. Yeah it is:)

**Pull request merged**